src/client/app/frameworks/core/decorators/page.component.ts

```diff
@@ -0,0 +1,13 @@
+import { DecoratorUtils } from './utils';
+import { Config } from '../utils/config';
+import { PageWrapService } from '../services/pagewrap.service';
+
+export function PageComponent(metadata: any = {}) {
+  return function(cls: any) {
+    return DecoratorUtils.annotateComponent(cls, metadata, {
+      providers: [
+        { provide: PageWrapService, useClass: Config.PageWrapService }
```

Comment:
Could you try using an OpaqueToken here?
I have a feeling that the per page token provider would work for page here. If it does then I think we can get rid of `PageWrapService` and `PageComponent` and just provide the page token on `BaseComponent` so any component would have the `PAGE` token to inject at their disposal when needed.